base/services/w32time/W32Time.c

```diff
@@ -0,0 +1,366 @@
+            break;
+        }
+    }
+
+    RegCloseKey(hKey);
+
+    ulTime = GetServerTime(szDefault);
+
+    if (ulTime != 0)
+        UpdateSystemTime(ulTime);
+
+    return 0;
+}
+
+// Control handler function
+VOID WINAPI
+ControlHandler(DWORD request) 
+{ 
+    switch(request) 
+    { 
+        case SERVICE_CONTROL_STOP: 
+            DPRINT("W32Time Service stopped.\n");
+
+            ServiceStatus.dwWin32ExitCode = 0; 
+            ServiceStatus.dwCurrentState  = SERVICE_STOPPED; 
+            SetServiceStatus (hStatus, &ServiceStatus);
+            return; 
+ 
+        case SERVICE_CONTROL_SHUTDOWN: 
+            DPRINT1("W32Time Service stopped.\n");
+
+            ServiceStatus.dwWin32ExitCode = 0; 
+            ServiceStatus.dwCurrentState  = SERVICE_STOPPED; 
+            SetServiceStatus (hStatus, &ServiceStatus);
+            return; 
+        
+        default:
+            break;
+    } 
+ 
+    // Report current status
+    SetServiceStatus (hStatus,  &ServiceStatus);
+ 
+    return; 
+}
+
+static VOID CALLBACK
+ServiceMain(DWORD argc, LPWSTR *argv)
+{
+    int   error;
+    int   result;
+    DWORD dwPollInterval;
+ 
+    UNREFERENCED_PARAMETER(argc);
+    UNREFERENCED_PARAMETER(argv);
+
+    ServiceStatus.dwServiceType             = SERVICE_WIN32; 
+    ServiceStatus.dwCurrentState            = SERVICE_START_PENDING; 
+    ServiceStatus.dwControlsAccepted        = SERVICE_ACCEPT_STOP | SERVICE_ACCEPT_SHUTDOWN;
+    ServiceStatus.dwWin32ExitCode           = 0; 
+    ServiceStatus.dwServiceSpecificExitCode = 0; 
+    ServiceStatus.dwCheckPoint              = 0; 
+    ServiceStatus.dwWaitHint                = 0; 
+ 
+    hStatus = RegisterServiceCtrlHandler(ServiceName,
+		                         ControlHandler);  // (LPHANDLER_FUNCTION)
+    if (hStatus == (SERVICE_STATUS_HANDLE)0) 
+    { 
+        // Registering Control Handler failed
+        return; 
+    }  
+    // Initialize Service 
+    error = InitService(); 
+    if (error) 
+    {
+        // Initialization failed
+        ServiceStatus.dwCurrentState       = SERVICE_STOPPED; 
+        ServiceStatus.dwWin32ExitCode      = -1; 
+        SetServiceStatus(hStatus, &ServiceStatus); 
+        return; 
+    } 
+    // We report the running status to SCM. 
+    ServiceStatus.dwCurrentState = SERVICE_RUNNING; 
+    SetServiceStatus (hStatus, &ServiceStatus);
+ 
+    // The worker loop of a service
+    while (ServiceStatus.dwCurrentState == SERVICE_RUNNING)
+        {
+            dwPollInterval = GetIntervalSetting();
+            result = SetTime();
+            if (result)
+            {
+                ServiceStatus.dwCurrentState       = SERVICE_STOPPED; 
+                ServiceStatus.dwWin32ExitCode      = -1; 
+                SetServiceStatus(hStatus, &ServiceStatus);
+                return;
+            }
+            Sleep(dwPollInterval);
+        }
+    return; 
+}
+
+int wmain(int argc, WCHAR *argv[]) 
+{
+
+    SERVICE_TABLE_ENTRYW ServiceTable[2] =
```

Comment:
No need to hardcode the number of elements (2) in the table here.

Reply:
I will take out the "2". Thanks.